auth/login.go

```diff
@@ -226,20 +226,36 @@ func login(ctx context.Context, newConfig *clientcmdapi.Config, kubeconfigPath,
 		format.PrintSuccessf("🏢", "switched to the organization %q", toOrg)
 	}
 	format.PrintSuccessf("📋", "added %s to kubeconfig", newConfig.CurrentContext)
+
+	authInfo, ok := newConfig.AuthInfos[newConfig.CurrentContext]
+	if !ok {
+		return fmt.Errorf("authInfo not found")
+	}
+
+	// We first try to read a static API token (e.g. APIToken parameter has been used)
+	token := authInfo.Token
+
 	if loginConfig.execPlugin {
-		authInfo := newConfig.AuthInfos[newConfig.CurrentContext]
 		if authInfo == nil || authInfo.Exec == nil {
 			return fmt.Errorf("no Exec found in authInfo")
 		}
 
-		// we discard the returned token as we just want to trigger the auth
-		// flow and populate the cache.
-		if _, err := api.GetTokenFromExecConfig(ctx, authInfo.Exec); err != nil {
+		// We just want to trigger the auth flow and populate the cache.
+		// We will also use the returned token to read UserInfo.
+		token, err = api.GetTokenFromExecConfig(ctx, authInfo.Exec)
+		if err != nil {
 			return err
 		}
+	}
 
+	userInfo, err := api.GetUserInfoFromToken(token)
+	if err != nil {
 		format.PrintSuccessf("🚀", "logged into cluster %s", newConfig.CurrentContext)
+
+		return nil
```

Comment:
Nit: Can you add a comment on why we don't return the error here? Are there any known cases where the user info can't be extracted from a valid token?

Reply:
yeah. It doesn't look good. I agree. However we have one real use case when there is no `UserInfo` on login available
```bash
nctl auth cluster <name>
# --exec-plugin=false by default
```

And even more cases in our unit tests:
1. auth/cluster_test.go
2. auth/login_test.go (TestLoginCmd, TestLoginCmdWithoutExistingKubeconfig)

assuming login() returns error on missing UserInfo, for the 2. case I was able to create the following workaround (added 'Organization` and `APIToken` to login command:
```golang
	// we run without the execPlugin, that would be something for an e2e test.
	// For this reason we also need valid Organization and APIToken fields to
	// pass GetUserInfoFromToken() in auth/login.go.
	cmd := &LoginCmd{
		ExecPlugin:   false,
		APIURL:       "https://" + apiHost,
		IssuerURL:    "https://auth.example.org",
		Organization: "test",
		APIToken:     FakeJWTToken,
	}
``` 
so this could work. 

But we still have `ClusterCmd` which accepts false for `ExecPlugin`, which means missing `UserInfo`, which means error on login.

Ignoring error on `GetUserInfoFromToken()` is the simplest solution that fixes everything. In the end, only information is missing - we do not use `UserInfo` for further calculations.

But maybe you have another idea how to solve this issue?

Reply:
as you suggested today, I simplified the login logic a bit. This solves all the problems described above.

Reply:
Looks good 👍